### ISSUE TITLE
Scenario locking does not work when scenarios are sorted by difficulty

### ISSUE THREAD

**Issue by Wuzzy2** [OpenRCT2/OpenRCT2#4009]
**OS:** GNU/Linux
**Version:** 0.0.4

If you sort scenarios by difficulty, the locked scenarios feature does not work and is disabled automatically. This does not make sense, this setting should be completely independent of the other.
- Reproducible in RCT2 (vanilla)? N/A
- Multiplayer? N/A

**Steps to reproduce:**
1. Start OpenRCT2 
2. Make sure you have not completed “too many scenarios” (or else you will have unlocked everything anyway)
3. Go to options
4. Sort scenarios by game
5. Enable scenario locking
6. Look into the scenario list and check the locked state of all scenarios
7. Now sort scenarios by difficulty in the options, don't touch anything else
8. Look into the scenario list again
9. Compare the locked state of the scenarios o steps 6 and 8

I expect: The same scenarios are locked in both sorting settings.
I get: When scenarios are sorted by difficulty, all scenarios are always unlocked.

But when sorting by difficulty, the unlocking of scenarios might be a bit counter-intuitive. That's okay, I simply suggest to change the hover text when hovering a locked scenario. It should be the same hover text, no matter the sorting.
Write something like “Complete 4 more scenarios in Wacky Worlds to unlock this scenario.”

**Comment by IntelOrca** [comment]
The way scenario locking is kind of iffy, it relies on the scenarios to be listed in the order that they must be completed in. So if we do enable locking scenarios when sorted by difficulty, it would force you to complete it in game order, rather than difficulty progression.

It was not implemented because of the above reason... (and it doesn't make sense to unlock them in a different order, particularly as you can switch between them later on). And more so this sorting never existed in RCT1 which had unlocking. RCT2 didn't.

**Comment by Wuzzy2** [comment]
Well but OpenRCT2 now has this feature. Since people decided to have this feature so it should be consistent in its implementation. It feels really dumb that a mere interface change suddenly unlocks all missions. How can you not see this as a bug? :P Would you rather have a broken locking feature than one that may be mildly irritating for 2 seconds?

> So if we do enable locking scenarios when sorted by difficulty, it would force you to complete it in game order, rather than difficulty progression.

The same is true when sorted by game, so this is besides the point. The real question is in what order we want the missions to have unlocked in the first place. Forget the sorting. Once decided, this should be the same no matter how scenarios are displayed. IMO we should stick by unlocking on a per-game basis for now because it is probably easier and more reliable. It's also how it worked in RCT + expansions. I have no idea how unlocking purely based on difficulty is even supposed to work, since different players can have different expansions (and thus: scenarios) installed. It would probably even more confusing to attempt to implement it and would probably not worth the hassle.

Long story short: Stick to the current system and make it consistent and independent from sorting.
